app/src/main/res/layout/tools_tutorial_banner.xml

```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <data>
+        <import type="androidx.databinding.ObservableField" />
+        <import type="org.cru.godtools.adapter.BannerCallbacks" />
+
+        <variable name="callbacks" type="ObservableField&lt;BannerCallbacks&gt;" />
+    </data>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="@dimen/tools_card_margin_horiz"
+        android:layout_marginRight="@dimen/tools_card_margin_horiz">
+
+        <androidx.appcompat.widget.AppCompatTextView
+            android:id="@+id/banner_open_description_text"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="16dp"
+            android:layout_marginRight="16dp"
+            android:firstBaselineToTopHeight="36dp"
```

Comment:
this attribute was added in API 28. it will be a long time until we can utilize it :(